### PR TITLE
 planner/core: fix unexpected error of window function

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

fix https://github.com/pingcap/tidb/issues/9554

### What is changed and how it works?

Let TiDB distinguish `over w` and `over(w)`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -87,3 +87,5 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20180531100431-4c381bd170b4
 	sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+replace gihtub.com/pingcap/parser => github.com/winoros/parser v0.0.0-20190305081723-e7cb6045c53f
```
Comment (eurekaka) [inline go.mod]: is this change related?

Comment (winoros) [reply go.mod]: Yes, this fix needs changes from parser.

**Comment by winoros** [comment]
@zz-jason We need to merge https://github.com/pingcap/parser/pull/236 to update go.mod

File: planner/core/logical_plan_builder.go [hunk 2]
```diff
@@ -2715,6 +2715,15 @@ func (b *PlanBuilder) buildProjectionForWindow(p LogicalPlan, expr *ast.WindowFu
 	b.optFlag |= flagEliminateProjection
 
 	var items []*ast.ByItem
+	if expr.Spec.Name.L != "" {
+		ref, ok := b.windowSpecs[expr.Spec.Name.L]
+		if !ok {
+			return nil, nil, nil, nil, ErrWindowNoSuchWindow.GenWithStackByArgs(expr.Spec.Name.O)
+		}
+		expr.Spec = ref
```
Comment (lamxTyler) [inline planner/core/logical_plan_builder.go]: Feels like we need a copy.
```sql
MySQL [test]> explain select row_number() over w, sum(a) over() w from t window w as (partition by a  rows between 1 preceding and 1 following);
+----------------------------+----------+------+------------------------------------------------------------+
| id                         | count    | task | operator info                                              |
+----------------------------+----------+------+------------------------------------------------------------+
| Projection_9               | 10000.00 | root | row_number() over w, w                                     |
| └─Window_10                | 10000.00 | root | sum(cast(test.t.a)) over()                                 |
|   └─Window_11              | 10000.00 | root | row_number() over(partition by test.t.a)                   |
|     └─Sort_15              | 10000.00 | root | test.t.a:asc                                               |
|       └─TableReader_14     | 10000.00 | root | data:TableScan_13                                          |
|         └─TableScan_13     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
+----------------------------+----------+------+------------------------------------------------------------+
```
The window for `sum(a)` is not correct.
